lib/jsdom/living/nodes/Element-impl.js

```diff
@@ -359,6 +315,51 @@ class ElementImpl extends NodeImpl {
     return 0;
   }
 
+  // https://dom.spec.whatwg.org/#dom-element-attachshadow
+  attachShadow(init) {
+    if (this.namespaceURI !== HTML_NS) {
+      throw new DOMException(
+        "This element does not support attachShadow.",
```

Comment:
Let's explain a bit more in the error messages, so that the two cases are distinct.

Reply:
I don't feel extremely inspired, let me know what you think about those error messages.